File: aidp-python-client/src/aidp_python_client/aidataplatform_dp/models/agent_flow_edge.py
# coding: utf-8
# Copyright (c) 2016, 2026, Oracle and/or its affiliates.  All rights reserved.
# This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.


from oci.util import formatted_flat_dict, NONE_SENTINEL, value_allowed_none_or_none_sentinel  # noqa: F401
from oci.decorators import init_model_state_from_kwargs


@init_model_state_from_kwargs
class AgentFlowEdge(object):
    """
    An edge connecting two nodes in the diagram, with context and metadata.
    """

    def __init__(self, **kwargs):
        """
        Initializes a new AgentFlowEdge object with values from keyword arguments.
        The following keyword arguments are supported (corresponding to the getters/setters of this class):

        :param src_node_id:
            The value to assign to the src_node_id property of this AgentFlowEdge.
        :type src_node_id: str

        :param dest_node_id:
            The value to assign to the dest_node_id property of this AgentFlowEdge.
        :type dest_node_id: str

        :param src_node_output:
            The value to assign to the src_node_output property of this AgentFlowEdge.
        :type src_node_output: str

        :param dest_node_input:
            The value to assign to the dest_node_input property of this AgentFlowEdge.
        :type dest_node_input: str

        :param parent_node_id:
            The value to assign to the parent_node_id property of this AgentFlowEdge.
        :type parent_node_id: str

        :param edge_handles:
            The value to assign to the edge_handles property of this AgentFlowEdge.
        :type edge_handles: list[oci.aidataplatform_dp.models.Point]

        :param edge_properties:
            The value to assign to the edge_properties property of this AgentFlowEdge.
        :type edge_properties: dict(str, object)

        :param edge_type:
            The value to assign to the edge_type property of this AgentFlowEdge.
        :type edge_type: str

        :param key:
            The value to assign to the key property of this AgentFlowEdge.
        :type key: str

        :param time_created:
            The value to assign to the time_created property of this AgentFlowEdge.
        :type time_created: datetime

        :param time_updated:
            The value to assign to the time_updated property of this AgentFlowEdge.
        :type time_updated: datetime

        :param validation_errors:
            The value to assign to the validation_errors property of this AgentFlowEdge.
        :type validation_errors: list[oci.aidataplatform_dp.models.ValidationError]

        """
        self.swagger_types = {
            'src_node_id': 'str',
            'dest_node_id': 'str',
            'src_node_output': 'str',
            'dest_node_input': 'str',
            'parent_node_id': 'str',
            'edge_handles': 'list[Point]',
            'edge_properties': 'dict(str, object)',
            'edge_type': 'str',
            'key': 'str',
            'time_created': 'datetime',
            'time_updated': 'datetime',
            'validation_errors': 'list[ValidationError]'
        }

        self.attribute_map = {
            'src_node_id': 'srcNodeId',
            'dest_node_id': 'destNodeId',
            'src_node_output': 'srcNodeOutput',
            'dest_node_input': 'destNodeInput',
            'parent_node_id': 'parentNodeId',
            'edge_handles': 'edgeHandles',
            'edge_properties': 'edgeProperties',
            'edge_type': 'edgeType',
            'key': 'key',
            'time_created': 'timeCreated',
            'time_updated': 'timeUpdated',
            'validation_errors': 'validationErrors'
        }

        self._src_node_id = None
        self._dest_node_id = None
        self._src_node_output = None
        self._dest_node_input = None
        self._parent_node_id = None
        self._edge_handles = None
        self._edge_properties = None
        self._edge_type = None
        self._key = None
        self._time_created = None
        self._time_updated = None
        self._validation_errors = None

    @property
    def src_node_id(self):
        """
        **[Required]** Gets the src_node_id of this AgentFlowEdge.
        Source node for this edge.


        :return: The src_node_id of this AgentFlowEdge.
        :rtype: str
        """
        return self._src_node_id

    @src_node_id.setter
    def src_node_id(self, src_node_id):
        """
        Sets the src_node_id of this AgentFlowEdge.
        Source node for this edge.


        :param src_node_id: The src_node_id of this AgentFlowEdge.
        :type: str
        """
        self._src_node_id = src_node_id

    @property
    def dest_node_id(self):
        """
        **[Required]** Gets the dest_node_id of this AgentFlowEdge.
        Destination node for this edge.


        :return: The dest_node_id of this AgentFlowEdge.
        :rtype: str
        """
        return self._dest_node_id

    @dest_node_id.setter
    def dest_node_id(self, dest_node_id):
        """
        Sets the dest_node_id of this AgentFlowEdge.
        Destination node for this edge.


        :param dest_node_id: The dest_node_id of this AgentFlowEdge.
        :type: str
        """
        self._dest_node_id = dest_node_id

    @property
    def src_node_output(self):
        """
        **[Required]** Gets the src_node_output of this AgentFlowEdge.
        Output port on the source node.


        :return: The src_node_output of this AgentFlowEdge.
        :rtype: str
        """
        return self._src_node_output

    @src_node_output.setter
    def src_node_output(self, src_node_output):
        """
        Sets the src_node_output of this AgentFlowEdge.
        Output port on the source node.


        :param src_node_output: The src_node_output of this AgentFlowEdge.
        :type: str
        """
        self._src_node_output = src_node_output

    @property
    def dest_node_input(self):
        """
        **[Required]** Gets the dest_node_input of this AgentFlowEdge.
        Input port on the destination node.


        :return: The dest_node_input of this AgentFlowEdge.
        :rtype: str
        """
        return self._dest_node_input

    @dest_node_input.setter
    def dest_node_input(self, dest_node_input):
        """
        Sets the dest_node_input of this AgentFlowEdge.
        Input port on the destination node.


        :param dest_node_input: The dest_node_input of this AgentFlowEdge.
        :type: str
        """
        self._dest_node_input = dest_node_input

    @property
    def parent_node_id(self):
        """
        Gets the parent_node_id of this AgentFlowEdge.
        Parent node, if hierarchical.


        :return: The parent_node_id of this AgentFlowEdge.
        :rtype: str
        """
        return self._parent_node_id

    @parent_node_id.setter
    def parent_node_id(self, parent_node_id):
        """
        Sets the parent_node_id of this AgentFlowEdge.
        Parent node, if hierarchical.


        :param parent_node_id: The parent_node_id of this AgentFlowEdge.
        :type: str
        """
        self._parent_node_id = parent_node_id

    @property
    def edge_handles(self):
        """
        Gets the edge_handles of this AgentFlowEdge.
        Geometry handle coordinates.


        :return: The edge_handles of this AgentFlowEdge.
        :rtype: list[oci.aidataplatform_dp.models.Point]
        """
        return self._edge_handles

    @edge_handles.setter
    def edge_handles(self, edge_handles):
        """
        Sets the edge_handles of this AgentFlowEdge.
        Geometry handle coordinates.


        :param edge_handles: The edge_handles of this AgentFlowEdge.
        :type: list[oci.aidataplatform_dp.models.Point]
        """
        self._edge_handles = edge_handles

    @property
    def edge_properties(self):
        """
        Gets the edge_properties of this AgentFlowEdge.
        Extensible Properties of the Edge


        :return: The edge_properties of this AgentFlowEdge.
        :rtype: dict(str, object)
        """
        return self._edge_properties

    @edge_properties.setter
    def edge_properties(self, edge_properties):
        """
        Sets the edge_properties of this AgentFlowEdge.
        Extensible Properties of the Edge


        :param edge_properties: The edge_properties of this AgentFlowEdge.
        :type: dict(str, object)
        """
        self._edge_properties = edge_properties

    @property
    def edge_type(self):
        """
        Gets the edge_type of this AgentFlowEdge.
        Style/type for edge.


        :return: The edge_type of this AgentFlowEdge.
        :rtype: str
        """
        return self._edge_type

    @edge_type.setter
    def edge_type(self, edge_type):
        """
        Sets the edge_type of this AgentFlowEdge.
        Style/type for edge.


        :param edge_type: The edge_type of this AgentFlowEdge.
        :type: str
        """
        self._edge_type = edge_type

    @property
    def key(self):
        """
        **[Required]** Gets the key of this AgentFlowEdge.
        Unique edge identifier.


        :return: The key of this AgentFlowEdge.
        :rtype: str
        """
        return self._key

    @key.setter
    def key(self, key):
        """
        Sets the key of this AgentFlowEdge.
        Unique edge identifier.


        :param key: The key of this AgentFlowEdge.
        :type: str
        """
        self._key = key

    @property
    def time_created(self):
        """
        Gets the time_created of this AgentFlowEdge.
        RFC3339 timestamp when edge was created.


        :return: The time_created of this AgentFlowEdge.
        :rtype: datetime
        """
        return self._time_created

    @time_created.setter
    def time_created(self, time_created):
        """
        Sets the time_created of this AgentFlowEdge.
        RFC3339 timestamp when edge was created.


        :param time_created: The time_created of this AgentFlowEdge.
        :type: datetime
        """
        self._time_created = time_created

    @property
    def time_updated(self):
        """
        Gets the time_updated of this AgentFlowEdge.
        RFC3339 timestamp when edge was last updated.


        :return: The time_updated of this AgentFlowEdge.
        :rtype: datetime
        """
        return self._time_updated

    @time_updated.setter
    def time_updated(self, time_updated):
        """
        Sets the time_updated of this AgentFlowEdge.
        RFC3339 timestamp when edge was last updated.


        :param time_updated: The time_updated of this AgentFlowEdge.
        :type: datetime
        """
        self._time_updated = time_updated

    @property
    def validation_errors(self):
        """
        Gets the validation_errors of this AgentFlowEdge.
        List of validation errors encountered in the diagram.


        :return: The validation_errors of this AgentFlowEdge.
        :rtype: list[oci.aidataplatform_dp.models.ValidationError]
        """
        return self._validation_errors

    @validation_errors.setter
    def validation_errors(self, validation_errors):
        """
        Sets the validation_errors of this AgentFlowEdge.
        List of validation errors encountered in the diagram.


        :param validation_errors: The validation_errors of this AgentFlowEdge.
        :type: list[oci.aidataplatform_dp.models.ValidationError]
        """
        self._validation_errors = validation_errors

    def __repr__(self):
        return formatted_flat_dict(self)

    def __eq__(self, other):
        if other is None:
            return False

        return self.__dict__ == other.__dict__

    def __ne__(self, other):
        return not self == other

File: aidp-java-client/src/main/java/com/oracle/aidataplatform/dp/model/AgentFlowEdge.java
package com.oracle.aidataplatform.dp.model;



/**
 * An edge connecting two nodes in the diagram, with context and metadata.
**/
@jakarta.annotation.Generated(value = "OracleSDKGenerator", comments = "API Version: 20260430")
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(builder=AgentFlowEdge.Builder.class)

public final class AgentFlowEdge  {
    @Deprecated
    @java.beans.ConstructorProperties({"srcNodeId", "destNodeId", "srcNodeOutput", "destNodeInput", "parentNodeId", "edgeHandles", "edgeProperties", "edgeType", "key", "timeCreated", "timeUpdated", "validationErrors"})
    public AgentFlowEdge(String srcNodeId, String destNodeId, String srcNodeOutput, String destNodeInput, String parentNodeId, java.util.List<Point> edgeHandles, java.util.Map<String, Object> edgeProperties, String edgeType, String key, java.util.Date timeCreated, java.util.Date timeUpdated, java.util.List<ValidationError> validationErrors) {
        super();
        this.srcNodeId = srcNodeId;
        this.destNodeId = destNodeId;
        this.srcNodeOutput = srcNodeOutput;
        this.destNodeInput = destNodeInput;
        this.parentNodeId = parentNodeId;
        this.edgeHandles = edgeHandles;
        this.edgeProperties = edgeProperties;
        this.edgeType = edgeType;
        this.key = key;
        this.timeCreated = timeCreated;
        this.timeUpdated = timeUpdated;
        this.validationErrors = validationErrors;
    }

    @com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder(withPrefix = "")
    public static class Builder {
                /**
     * Source node for this edge.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("srcNodeId")
private String srcNodeId;

        /**
         * Source node for this edge.
         * @param srcNodeId the value to set
         * @return this builder
         **/
        

public Builder srcNodeId(String srcNodeId) {
    this.srcNodeId = srcNodeId;
    return this;
}
            /**
     * Destination node for this edge.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("destNodeId")
private String destNodeId;

        /**
         * Destination node for this edge.
         * @param destNodeId the value to set
         * @return this builder
         **/
        

public Builder destNodeId(String destNodeId) {
    this.destNodeId = destNodeId;
    return this;
}
            /**
     * Output port on the source node.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("srcNodeOutput")
private String srcNodeOutput;

        /**
         * Output port on the source node.
         * @param srcNodeOutput the value to set
         * @return this builder
         **/
        

public Builder srcNodeOutput(String srcNodeOutput) {
    this.srcNodeOutput = srcNodeOutput;
    return this;
}
            /**
     * Input port on the destination node.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("destNodeInput")
private String destNodeInput;

        /**
         * Input port on the destination node.
         * @param destNodeInput the value to set
         * @return this builder
         **/
        

public Builder destNodeInput(String destNodeInput) {
    this.destNodeInput = destNodeInput;
    return this;
}
            /**
     * Parent node, if hierarchical.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("parentNodeId")
private String parentNodeId;

        /**
         * Parent node, if hierarchical.
         * @param parentNodeId the value to set
         * @return this builder
         **/
        

public Builder parentNodeId(String parentNodeId) {
    this.parentNodeId = parentNodeId;
    return this;
}
            /**
     * Geometry handle coordinates.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("edgeHandles")
private java.util.List<Point> edgeHandles;

        /**
         * Geometry handle coordinates.
         * @param edgeHandles the value to set
         * @return this builder
         **/
        

public Builder edgeHandles(java.util.List<Point> edgeHandles) {
    this.edgeHandles = edgeHandles;
    return this;
}
            /**
     * Extensible Properties of the Edge
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("edgeProperties")
private java.util.Map<String, Object> edgeProperties;

        /**
         * Extensible Properties of the Edge
         * @param edgeProperties the value to set
         * @return this builder
         **/
        

public Builder edgeProperties(java.util.Map<String, Object> edgeProperties) {
    this.edgeProperties = edgeProperties;
    return this;
}
            /**
     * Style/type for edge.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("edgeType")
private String edgeType;

        /**
         * Style/type for edge.
         * @param edgeType the value to set
         * @return this builder
         **/
        

public Builder edgeType(String edgeType) {
    this.edgeType = edgeType;
    return this;
}
            /**
     * Unique edge identifier.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("key")
private String key;

        /**
         * Unique edge identifier.
         * @param key the value to set
         * @return this builder
         **/
        

public Builder key(String key) {
    this.key = key;
    return this;
}
            /**
     * RFC3339 timestamp when edge was created.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("timeCreated")
private java.util.Date timeCreated;

        /**
         * RFC3339 timestamp when edge was created.
         * @param timeCreated the value to set
         * @return this builder
         **/
        

public Builder timeCreated(java.util.Date timeCreated) {
    this.timeCreated = timeCreated;
    return this;
}
            /**
     * RFC3339 timestamp when edge was last updated.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("timeUpdated")
private java.util.Date timeUpdated;

        /**
         * RFC3339 timestamp when edge was last updated.
         * @param timeUpdated the value to set
         * @return this builder
         **/
        

public Builder timeUpdated(java.util.Date timeUpdated) {
    this.timeUpdated = timeUpdated;
    return this;
}
            /**
     * List of validation errors encountered in the diagram.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("validationErrors")
private java.util.List<ValidationError> validationErrors;

        /**
         * List of validation errors encountered in the diagram.
         * @param validationErrors the value to set
         * @return this builder
         **/
        

public Builder validationErrors(java.util.List<ValidationError> validationErrors) {
    this.validationErrors = validationErrors;
    return this;
}


        public AgentFlowEdge build() {
            AgentFlowEdge model = new AgentFlowEdge(this.srcNodeId
                , this.destNodeId
                , this.srcNodeOutput
                , this.destNodeInput
                , this.parentNodeId
                , this.edgeHandles
                , this.edgeProperties
                , this.edgeType
                , this.key
                , this.timeCreated
                , this.timeUpdated
                , this.validationErrors);            return model;
        }

        @com.fasterxml.jackson.annotation.JsonIgnore
        public Builder copy(AgentFlowEdge model) {
                this.srcNodeId(model.getSrcNodeId());
    this.destNodeId(model.getDestNodeId());
    this.srcNodeOutput(model.getSrcNodeOutput());
    this.destNodeInput(model.getDestNodeInput());
    this.parentNodeId(model.getParentNodeId());
    this.edgeHandles(model.getEdgeHandles());
    this.edgeProperties(model.getEdgeProperties());
    this.edgeType(model.getEdgeType());
    this.key(model.getKey());
    this.timeCreated(model.getTimeCreated());
    this.timeUpdated(model.getTimeUpdated());
    this.validationErrors(model.getValidationErrors());
return this;
        }
    }

    /**
     * Create a new builder.
     */
    public static Builder builder() {
        return new Builder();
    }


    public Builder toBuilder() {
        return new Builder().copy(this);
    }

    


        /**
     * Source node for this edge.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("srcNodeId")
    private final String srcNodeId;

        /**
     * Source node for this edge.
     * @return the value
     **/
    
    public String getSrcNodeId() {
        return srcNodeId;
    }


        /**
     * Destination node for this edge.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("destNodeId")
    private final String destNodeId;

        /**
     * Destination node for this edge.
     * @return the value
     **/
    
    public String getDestNodeId() {
        return destNodeId;
    }


        /**
     * Output port on the source node.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("srcNodeOutput")
    private final String srcNodeOutput;

        /**
     * Output port on the source node.
     * @return the value
     **/
    
    public String getSrcNodeOutput() {
        return srcNodeOutput;
    }


        /**
     * Input port on the destination node.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("destNodeInput")
    private final String destNodeInput;

        /**
     * Input port on the destination node.
     * @return the value
     **/
    
    public String getDestNodeInput() {
        return destNodeInput;
    }


        /**
     * Parent node, if hierarchical.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("parentNodeId")
    private final String parentNodeId;

        /**
     * Parent node, if hierarchical.
     * @return the value
     **/
    
    public String getParentNodeId() {
        return parentNodeId;
    }


        /**
     * Geometry handle coordinates.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("edgeHandles")
    private final java.util.List<Point> edgeHandles;

        /**
     * Geometry handle coordinates.
     * @return the value
     **/
    
    public java.util.List<Point> getEdgeHandles() {
        return edgeHandles;
    }


        /**
     * Extensible Properties of the Edge
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("edgeProperties")
    private final java.util.Map<String, Object> edgeProperties;

        /**
     * Extensible Properties of the Edge
     * @return the value
     **/
    
    public java.util.Map<String, Object> getEdgeProperties() {
        return edgeProperties;
    }


        /**
     * Style/type for edge.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("edgeType")
    private final String edgeType;

        /**
     * Style/type for edge.
     * @return the value
     **/
    
    public String getEdgeType() {
        return edgeType;
    }


        /**
     * Unique edge identifier.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("key")
    private final String key;

        /**
     * Unique edge identifier.
     * @return the value
     **/
    
    public String getKey() {
        return key;
    }


        /**
     * RFC3339 timestamp when edge was created.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("timeCreated")
    private final java.util.Date timeCreated;

        /**
     * RFC3339 timestamp when edge was created.
     * @return the value
     **/
    
    public java.util.Date getTimeCreated() {
        return timeCreated;
    }


        /**
     * RFC3339 timestamp when edge was last updated.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("timeUpdated")
    private final java.util.Date timeUpdated;

        /**
     * RFC3339 timestamp when edge was last updated.
     * @return the value
     **/
    
    public java.util.Date getTimeUpdated() {
        return timeUpdated;
    }


        /**
     * List of validation errors encountered in the diagram.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("validationErrors")
    private final java.util.List<ValidationError> validationErrors;

        /**
     * List of validation errors encountered in the diagram.
     * @return the value
     **/
    
    public java.util.List<ValidationError> getValidationErrors() {
        return validationErrors;
    }

    @Override
    public String toString() {
        return this.toString(true);
    }

    /**
     * Return a string representation of the object.
     * @param includeByteArrayContents true to include the full contents of byte arrays
     * @return string representation
     */
    public String toString(boolean includeByteArrayContents) {
        java.lang.StringBuilder sb = new java.lang.StringBuilder();
        sb.append("AgentFlowEdge(");
        sb.append("srcNodeId=").append(String.valueOf(this.srcNodeId));
        sb.append(", destNodeId=").append(String.valueOf(this.destNodeId));
        sb.append(", srcNodeOutput=").append(String.valueOf(this.srcNodeOutput));
        sb.append(", destNodeInput=").append(String.valueOf(this.destNodeInput));
        sb.append(", parentNodeId=").append(String.valueOf(this.parentNodeId));
        sb.append(", edgeHandles=").append(String.valueOf(this.edgeHandles));
        sb.append(", edgeProperties=").append(String.valueOf(this.edgeProperties));
        sb.append(", edgeType=").append(String.valueOf(this.edgeType));
        sb.append(", key=").append(String.valueOf(this.key));
        sb.append(", timeCreated=").append(String.valueOf(this.timeCreated));
        sb.append(", timeUpdated=").append(String.valueOf(this.timeUpdated));
        sb.append(", validationErrors=").append(String.valueOf(this.validationErrors));
        sb.append(")");
        return sb.toString();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof AgentFlowEdge)) {
            return false;
        }

        AgentFlowEdge other = (AgentFlowEdge) o;
        return java.util.Objects.equals(this.srcNodeId, other.srcNodeId) &&
            java.util.Objects.equals(this.destNodeId, other.destNodeId) &&
            java.util.Objects.equals(this.srcNodeOutput, other.srcNodeOutput) &&
            java.util.Objects.equals(this.destNodeInput, other.destNodeInput) &&
            java.util.Objects.equals(this.parentNodeId, other.parentNodeId) &&
            java.util.Objects.equals(this.edgeHandles, other.edgeHandles) &&
            java.util.Objects.equals(this.edgeProperties, other.edgeProperties) &&
            java.util.Objects.equals(this.edgeType, other.edgeType) &&
            java.util.Objects.equals(this.key, other.key) &&
            java.util.Objects.equals(this.timeCreated, other.timeCreated) &&
            java.util.Objects.equals(this.timeUpdated, other.timeUpdated) &&
            java.util.Objects.equals(this.validationErrors, other.validationErrors);
    }

    @Override
    public int hashCode() {
        final int PRIME = 59;
        int result = 1;
        result = (result * PRIME) + (this.srcNodeId == null ? 43 : this.srcNodeId.hashCode());
        result = (result * PRIME) + (this.destNodeId == null ? 43 : this.destNodeId.hashCode());
        result = (result * PRIME) + (this.srcNodeOutput == null ? 43 : this.srcNodeOutput.hashCode());
        result = (result * PRIME) + (this.destNodeInput == null ? 43 : this.destNodeInput.hashCode());
        result = (result * PRIME) + (this.parentNodeId == null ? 43 : this.parentNodeId.hashCode());
        result = (result * PRIME) + (this.edgeHandles == null ? 43 : this.edgeHandles.hashCode());
        result = (result * PRIME) + (this.edgeProperties == null ? 43 : this.edgeProperties.hashCode());
        result = (result * PRIME) + (this.edgeType == null ? 43 : this.edgeType.hashCode());
        result = (result * PRIME) + (this.key == null ? 43 : this.key.hashCode());
        result = (result * PRIME) + (this.timeCreated == null ? 43 : this.timeCreated.hashCode());
        result = (result * PRIME) + (this.timeUpdated == null ? 43 : this.timeUpdated.hashCode());
        result = (result * PRIME) + (this.validationErrors == null ? 43 : this.validationErrors.hashCode());
        return result;
    }


}
